scan-command/src/main/java/io/ballerina/scan/utils/ScanUtils.java

```diff
@@ -276,6 +301,242 @@ private static void unzipReportResources(InputStream source, File target) throws
+     * @return true if the file is a local file, false otherwise
+     */
+    private static boolean isLocalFile(Optional<Path> path) throws InvalidPathException {
+        return path.isPresent() && new File(path.get().toString()).exists();
+    }
+
+    /**
+     * Returns the {@link ScanTomlFile} representation a remote Scan.toml file from the provided URL.
+     *
+     * @param targetDir              the target directory
+     * @param remoteScanTomlFilePath the remote Scan.toml file path
+     * @param outputStream           the print stream
+     * @return an in-memory representation of the remote Scan.toml file
+     */
+    private static ScanTomlFile loadRemoteScanFile(Path targetDir, URL remoteScanTomlFilePath,
+                                                   PrintStream outputStream) {
+        Path cachePath = targetDir.resolve(REPORT_DIR_NAME).resolve(SCAN_FILE);
+
+        if (Files.exists(cachePath)) {
+            outputStream.println("Loading scan tool configurations from cache...");
+            return loadScanFile(targetDir, cachePath, outputStream);
+        }
+
+        try {
+            FileUtils.copyURLToFile(remoteScanTomlFilePath, cachePath.toFile());
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to download remote configuration file with exception: "
+                    + ex.getMessage());
+        }
+        outputStream.println("Loading scan tool configurations from " + remoteScanTomlFilePath.toString());
+        return loadScanFile(targetDir, cachePath, outputStream);
+    }
+
+    /**
+     * Returns the {@link ScanTomlFile} representation of the Scan.toml file.
+     *
+     * @param targetDir        the target directory
+     * @param scanTomlFilePath the local Scan.toml file path
+     * @param outputStream     the print stream
+     * @return an in-memory representation of the Scan.toml file
+     */
+    private static ScanTomlFile loadScanFile(Path targetDir, Path scanTomlFilePath, PrintStream outputStream) {
+        Toml scanTomlDocumentContent;
+
+        try {
+            scanTomlDocumentContent = Toml.read(scanTomlFilePath);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to read the configuration file with exception: " + ex.getMessage());
+        }
+        ScanTomlFile scanTomlFile = new ScanTomlFile();
+
+        List<Toml> platformTables = scanTomlDocumentContent.getTables(PLATFORM_TABLE);
+        platformTables.forEach(platformTable -> {
+            Map<String, Object> properties = platformTable.toMap();
+            String name = !(properties.get("name") instanceof String) ? null :
+                    properties.remove("name").toString();
+            String path = !(properties.get("path") instanceof String) ? null :
+                    properties.remove("path").toString();
+
+            if (name != null && !name.isEmpty() && path != null) {
+                if (!(new File(path).exists())) {
```

Comment:
We can combine these two conditions

Reply:
This is not possible, as the program should check for the path to the JAR file only if the name and attributes are provided in the `Scan.toml` file. Furthermore, if the provided path is to a remote JAR, it should try to download it first and then assign the downloaded JARs path.

Will inverse the conditions and do an early return to address this.